src/core/control/tools/EraseHandler.cpp

```diff
@@ -55,10 +55,9 @@ void EraseHandler::erase(double x, double y) {
     for (Element* e: l->getElements()) {
         if (e->getType() == ELEMENT_STROKE && e->intersectsArea(&eraserRect)) {
             eraseStroke(l, dynamic_cast<Stroke*>(e), x, y, range);
+            this->view->elementChanged(e);
```

Comment:
Actually, I think you augmented the rerendered range here. Before, only the required range was rerendered:

1. In "delete stroke mode", if a stroke was actually untouched by the eraser, it's bounding box was not added to the range.
2. In "normal" mode, only the square below the eraser was rerendered.

Reply:
Mmm, that's weird. Either the range is not large enough (I don't see why), or the repainted rectangle is too small when a Range is passed to view->rerenderRange().

Reply:
I'm currently leaning towards this theory: some amount of padding is necessary because of rounding errors and maybe anti-aliasing.

The rounding errors would come from the two different ways those coordinates are computed: using cairo internals when painting, and using our theoretical bounds otherwise. Two different computation methods would lead to (ever so slightly) different results. Not sure this could actually have a noticeable effect.

Anti-aliasing would make cairo paint stlightly outside of the theoretical bounds, thus our theoretical bound need to be padded when rerendering/repainting.

I'm not so sure about all this, so I need to think a bit more.

Reply:
Yes, I agree. The only thing: the padding should be located in a single function (I guess rerenderRect - so remove the padding in rerenderElement) and the size of the padding should be a constexpr.